common/socket/src/main/java/io/helidon/common/socket/IdleInputStream.java

```diff
@@ -107,20 +122,46 @@ boolean isClosed() {
 
     private void handle() {
         try {
-            next = upstream.read();
-            if (next <= 0) {
-                closed = true;
+            //Currently configured socket read timeout. This is intended to be restored after this method finishes
+            int toRestore = -1;
+            for (int i = 0; !cancelled; i++) {
+                try {
+                    toRestore = timeoutToRestore(toRestore);
```

Comment:
No need for a method, having the if statement here makes it more readable.

Reply:
Hm... I see this as a cleaner option :-)

Reply:
How is this better, it's 3 lines of code, it's used only once and does not mutate any state.

Reply:
I agree with Romain, that should be inlined. Also assigning to a param is generally frowned upon.

Reply:
You mean to use ternary operator? What is so bothering you guys in having this method?

Yeah, that parameter assignment was a mistake. I did not realize it is a parameter. It is fixed in the latest PR already.

Reply:
Because it's just one if statement with an assignment, that shouldn't be in a method.

```java
int t = socket.getSoTimeout();
if (t != ITERATION_TIME_MILLIS) {
    toRestore = t;
}
```

Reply:
If this method bothers you so much, I will remove it.

Reply:
Alright, I will remove it, if it helps to get this merged.